utils/wgs_gene_coverage/get_gene_exon_intron.R

```diff
@@ -0,0 +1,42 @@
+library(data.table)
+library(rtracklayer)
+library(base)
+
+## Read genome intron/exon coordaintes (from parse_gft.R)
+genome_bed <- fread("/hpf/largeprojects/ccm_dccforge/dccdipg/Common/genomes/hg19_introns_exons.bed")
+
+
+Get_Gene_Bed <- function(gene_name, dir){
+  ## Print input gene_name
+  print(paste("Input gene:", gene_name))
+  
+  ## Check if gene_name input is valid
+  valid_genes <- unique(genome_bed$gene_id)
+  if (! gene_name %in% valid_genes){
+    stop("Invalid gene name")
+  }
+  
+  ## Filter genome_bed for gene reigons 
+  gene_bed <- genome_bed[which(genome_bed$gene_id == gene_name),]
+
+  ## Concatenate annotation columns into "id" (mosdepth only allows one metadata column)
+  gene_bed$id <- paste(gene_bed$transcript_id, gene_bed$gene_id, gene_bed$type, sep=";") 
+  gene_bed <- subset(gene_bed, select = -c(transcript_id, gene_id, type))
+
+  write.table(gene_bed, paste(dir, "gene_coverage/", sprintf("%s_pos.bed", gene_name), sep=""), ######## TODO - check path 
```

Comment:
Need a line to create ${dir}/gene_coverage because the folder does not exist, and the script throws an error. You could add that here before writing the table, or create the directory in the get_gene_coverage.sh script before running this script.